libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/Imp/QuerySpec.hs

```diff
@@ -0,0 +1,249 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Test.Cardano.Ledger.Api.State.Imp.QuerySpec where
+
+import Cardano.Ledger.Api.State.Query (
+  CommitteeMemberState (..),
+  CommitteeMembersState (..),
+  HotCredAuthStatus (..),
+  MemberStatus (..),
+  NextEpochChange (..),
+  queryCommitteeMembersState,
+ )
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Coin (Coin (Coin))
+import Cardano.Ledger.Conway.Governance (
+  Committee (..),
+  ConwayGovState,
+  EraGov (..),
+  GovAction (..),
+  GovActionPurpose (..),
+  PrevGovActionId (..),
+  Voter (DRepVoter),
+  cgEnactStateL,
+  ensCommitteeL,
+ )
+import Cardano.Ledger.Conway.PParams (
+  dvtCommitteeNoConfidence,
+  dvtCommitteeNormal,
+  dvtUpdateToConstitution,
+  ppCommitteeMaxTermLengthL,
+  ppDRepVotingThresholdsL,
+  ppGovActionDepositL,
+  ppGovActionLifetimeL,
+ )
+import Cardano.Ledger.Core
+import Cardano.Ledger.Credential (Credential (KeyHashObj))
+import Cardano.Ledger.Keys (
+  KeyHash,
+  KeyRole (..),
+ )
+import Cardano.Ledger.Shelley.LedgerState
+import Control.Monad (replicateM)
+import Data.Default.Class (Default (..))
+import Data.Foldable (Foldable (..))
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+import Lens.Micro ((&), (.~))
+import Lens.Micro.Mtl
+import Test.Cardano.Ledger.Conway.ImpTest
+import Test.Cardano.Ledger.Core.Rational (IsRatio (..))
+import Test.Cardano.Ledger.Imp.Common
+
+spec ::
+  forall era.
+  ( HasCallStack
+  , ConwayEraImp era
+  , GovState era ~ ConwayGovState era
+  ) =>
+  SpecWith (ImpTestState era)
+spec =
+  it "Committee queries" $ do
+    setPParams
+    drep <- setupSingleDRep
+    vals <- replicateM 8 freshKeyHash
+    let (c1, c2, c3, c4, c5, c6, c7, c8) = case vals of
+          [c1', c2', c3', c4', c5', c6', c7', c8'] -> (c1', c2', c3', c4', c5', c6', c7', c8')
```

Comment:
This is so unpleasant - is there a better way?

Reply:
Sure.
```haskell
c1 <- freshKeyHash
c2 <- freshKeyHash
c3 <- freshKeyHash
c4 <- freshKeyHash
c5 <- freshKeyHash
c6 <- freshKeyHash
c7 <- freshKeyHash
```

Reply:
Haha! Fair enough